fenicsprecice/fenicsprecice.py

```diff
@@ -54,6 +54,10 @@ def __init__(self, adapter_config_filename='precice-adapter-config.json'):
         self._rank = self._comm.Get_rank()
         self._size = self._comm.Get_size()
 
+        self._is_parallel = True
+        if self._size == 1:
+            self._is_parallel = False
```

Comment:
Better create a (private) function:

```
def _is_parallel():
    return self._size > 1
```

This makes the class more safe, since it ensures that `self._is_parallel` and `self._size` are consistent. Maybe, even replace `self._size` and `self._rank` with `self._comm.Get_rank()` and `self._comm.Get_size()` (as long as this is not performance critical).